tripy/tripy/frontend/module/sequential.py

```diff
@@ -0,0 +1,185 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Union, Tuple, Dict, Iterator, Any
+from dataclasses import dataclass
+import copy
+
+from tripy import export
+from tripy.common.exception import raise_error
+from tripy.frontend.module import Module
+
+
+@export.public_api(
+    document_under="modules/sequential.rst",
+    autodoc_options=[
+        ":exclude-members: __getattr__",
+    ],
+)
+@dataclass
+class Sequential(Module):
+    r"""
+    A module to stack multiple layers or modules in a sequential order. The `Sequential`
+    container can accept either a list of modules or a dictionary of named modules. Modules are
+    added in the order they are passed, and each is called sequentially during the forward pass.
+    """
+
+    def __init__(self, *modules: Union[Module, Dict[str, Module]]) -> None:
+        r"""
+        Args:
+            *modules: The modules to include in the sequence.
+                Can be passed as individual positional arguments or as a single dictionary of named modules.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Sequential with Positional Arguments
+
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+
+            input = tp.Tensor([1.0])
+            output = model(input)
+
+        .. code-block:: python
+            :linenos:
+            :caption: Sequential with a Dictionary
+
+            model = tp.Sequential({'layer1': tp.Linear(1, 3), 'layer2': tp.Linear(3, 2)})
+
+            input = tp.Tensor([1.0])
+            output = model(input)
+        """
+        super().__init__()
+        self.modules = {}
+
+        if len(modules) == 1 and isinstance(modules[0], dict):
+            self.modules = copy.copy(modules[0])
+        else:
+            for idx, module in enumerate(modules):
+                self.modules[str(idx)] = module
+
+    def __call__(self, input: "tripy.Tensor") -> "tripy.Tensor":
+        r"""
+        Defines the forward pass by applying each module in the container sequentially to `input`
+
+        Args:
+            input: The input tensor to pass through the sequence of modules.
+
+        Returns:
+            The output tensor after passing through each module in sequence.
+        """
+        for module in self.modules.values():
+            input = module(input)
+        return input
+
+    def __getattr__(self, name) -> Any:
+        """
+        Custom __getattr__ to search both in `modules` dictionary and in other attributes. This is for handling
+        `module = operator.attrgetter(child_name)(module)` calls in tripy/frontend/module/module.py:load_state_dict
+        """
+        # Check if `name` is a key in the modules dictionary
+        if name in self.modules:
+            return self.modules[name]
```

Comment:
Does this also work for integers? I'm wondering if it would be better to make `Sequential` a subclass of `dict` as well. That would also allow us to drop len/iter/getitem below.

Reply:
hmmm I rather not, since dict has other methods that we don't necessarily want Sequential to have like, `__setattr__` after construction.

Also, subclass of dict and Module would be multiple inheritance and could be a problem in case Module and dict share any mutual methods which could introduce hidden bugs for Sequential.

Reply:
Instead i can just cast name to str and add typing to name `name: Union[str, int]`

Reply:
But how can you call `__getattr__` with an integer? Doing something like `module.0` wouldn't be valid syntax

Reply:
you are right. Should we just allow strings then?

Reply:
That's why I was suggesting inheriting from `dict`; we have special handling that will bypass the `operator.attrgetter` for dictionaries. Should also work to only allow strings though, especially if we don't expect this to be user facing.

Reply:
yeah I see. I added the string type to `name` and yes since it does not face users, I think it will work.

But I still don't like inheriting from dict, since we will inherit other methods that we don't necessarily need or want.